metricbeat/module/kafka/partition/partition.go

```diff
@@ -322,3 +320,111 @@ func checkRetryQuery(err error) (retry, reconnect bool) {
 
 	return false, false
 }
+
+func findMatchingBroker(
+	addr string,
+	advertised string,
+	brokers []*sarama.Broker,
+) *sarama.Broker {
+	// compare connection address to list of broker addresses
+	for _, b := range brokers {
+		if advertised != "" && b.Addr() == advertised {
+			return b
+		}
+
+		if b.Addr() == addr {
```

Comment:
`b.Addr()` will be the address we configured (defaults to `localhost:9092`). the `meta.Brokers` will contains the addressed as advertised by kafka broker itself. Might be different.